src/pymor/models/iosys.py

```diff
@@ -655,43 +713,59 @@ def gramian(self, typ, mu=None):
         gramian = self.presets[typ] if typ in self.presets else self._gramian(typ, mu=mu)
 
         # assert correct return types
-        if typ == 'c_lrcf':
-            assert gramian in self.A.source
-        elif typ == 'o_lrcf':
-            assert gramian in self.A.range
-        elif typ == 'c_dense':
-            assert isinstance(gramian, np.ndarray) and gramian.shape == (self.A.source.dim, self.A.range.dim)
-        elif typ == 'o_dense':
-            assert isinstance(gramian, np.ndarray) and gramian.shape == (self.A.range.dim, self.A.source.dim)
+        assert ((isinstance(typ, str) and typ.endswith('_lrcf') or isinstance(typ, tuple))
+                and gramian in self.A.source
+                or isinstance(gramian, np.ndarray)
+                and gramian.shape == (self.A.source.dim, self.A.source.dim))
 
         return gramian
 
     @cached
-    def _hsv_U_V(self, mu=None):
-        """Compute Hankel singular values and vectors.
+    def _sv_U_V(self, typ='lyap', mu=None):
+        """Compute (Hankel) singular values and vectors.
 
         .. note::
             Assumes the system is asymptotically stable.
 
         Parameters
         ----------
+        typ
+            The type of the Gramians used:
+
+            - `'lyap'`: Lyapunov Gramian,
+            - `'bs'`: Bernoulli stabilized Gramian,
+            - `'lqg'`: LQG Gramian,
+            - `('br', gamma)`: bounded real Gramian,
         mu
             |Parameter values|.
 
         Returns
         -------
-        hsv
+        sv
             One-dimensional |NumPy array| of singular values.
         Uh
-            |NumPy array| of left singular vectors.
+            |NumPy array| of left singular vectors as rows.
         Vh
-            |NumPy array| of right singular vectors.
+            |NumPy array| of right singular vectors as rows.
         """
         if not isinstance(mu, Mu):
             mu = self.parameters.parse(mu)
         assert self.parameters.assert_compatible(mu)
-        cf = self.gramian('c_lrcf', mu=mu)
-        of = self.gramian('o_lrcf', mu=mu)
+        if typ == 'lyap':
+            cf = self.gramian('c_lrcf', mu=mu)
+            of = self.gramian('o_lrcf', mu=mu)
+        elif typ == 'bs':
+            cf = self.gramian('bs_c_lrcf', mu=mu)
+            of = self.gramian('bs_o_lrcf', mu=mu)
+        elif typ == 'lqg':
+            cf = self.gramian('lqg_c_lrcf', mu=mu)
+            of = self.gramian('lqg_o_lrcf', mu=mu)
+        elif isinstance(typ, tuple) and typ[0] == 'br' and typ[1] > 0:
+            gamma = typ[1]
+            cf = self.gramian(('br_c_lrcf', gamma), mu=mu)
+            of = self.gramian(('br_o_lrcf', gamma), mu=mu)
+        else:
+            raise ValueError(f'Unknown typ ({typ}).')
         U, hsv, Vh = spla.svd(self.E.apply2(of, cf, mu=mu), lapack_driver='gesvd')
         return hsv, U.T, Vh
```

Comment:
You're right, this would not work as expected for complex-valued systems. There are probably more places where things break down (I don't even know if there is literature on BT or IRKA for complex-valued systems). But, I think this is a separate issue from the PR, so feel free to open a new issue or discussion about this.